mlir/lib/Gradient/Transforms/ConversionPatterns.cpp

```diff
@@ -375,6 +493,162 @@ struct BackpropOpPattern : public ConvertOpToLLVMPattern<BackpropOp> {
         return bufferSizeBytes;
     }
 
+    void convertCustomGradArgumentTypes(TypeRange argTypes,
+                                        SmallVectorImpl<Type> &llvmArgTypes) const
+    {
+        for (Type argType : argTypes) {
+            llvmArgTypes.append(2, argType);
+        }
+    }
+
+    func::FuncOp genAugmentedForward(func::FuncOp qnode, OpBuilder &builder) const
```

Comment:
For maintenance, it would be great to define what is a augmented forward pass or to link to enzyme doc.